contracts/utility/ExpDecayMath.sol

```diff
@@ -12,12 +12,13 @@ library ExpDecayMath {
      * @dev returns the amount required for a token after a given time period since trading has been enabled
      *
      * the returned value is calculated as `amount / 2 ^ (timeElapsed / halfLife)`
-     * note that because the exponentiation function is limited to an input of up to (and excluding)
-     * 16 / ln 2, the input value to this function is limited by `timeElapsed / halfLife < 16 / ln 2`
+     * note that the input value to this function is limited by `timeElapsed / halfLife <= 128`
      */
     function calcExpDecay(uint256 amount, uint32 timeElapsed, uint32 halfLife) internal pure returns (uint256) {
-        Fraction memory input = Fraction({ n: timeElapsed, d: halfLife });
+        uint256 integerPart = timeElapsed / halfLife;
```

Comment:
The more accurate condition to assert (directly on the input) is:
```
timeElapsed < halfLife * 129
```
But note that this restriction is tightly coupled with the implementation of the `exp2` function.
I therefore did not enforce it in the `expDecay` function (just like we don't normally impose these checks on function input in most cases, and leave it for the EVM to handle during runtime).
Let me know if you prefer otherwise (will require adapting the expected error-message in the tests).

Also note that `halfLife * 129` by itself can overflow, so the exact condition is technically:
```
timeElapsed < uint256(halfLife) * 129
```
Of course, when `halfLife * 129` overflows, the expression `timeElapsed < uint256(halfLife) * 129` is false.

Reply:
on second thought, might not be necessary - might be better to leave it as is